stable/ambassador/templates/deployment.yaml

```
@@ -136,6 +136,30 @@ spec:
           {{- end }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+        {{- if .Values.pro.enabled }}
+        - name: ambassador-pro
+          image: "{{ .Values.pro.image }}"
+          ports:
+          - name: grpc-auth
+            containerPort: {{ .Values.pro.ports.auth }}
+          - name: grpc-ratelimit
+            containerPort: {{ .Values.pro.ports.ratelimit }}
+          - name: http-debug
+            containerPort: {{ .Values.pro.ports.ratelimitDebug }}
+          env:
+          - name: REDIS_SOCKET_TYPE
+            value: tcp
+          - name: REDIS_URL
+            value: ambassador-pro-redis:6379
+          - name: APRO_AUTH_PORT
+            value: "{{ .Values.pro.ports.auth }}"
+          - name: GRPC_PORT
+            value: "{{ .Values.pro.ports.ratelimit }}"
+          - name: DEBUG_PORT
+            value: "{{ .Values.pro.ports.ratelimitDebug }}"
+          - name: AMBASSADOR_LICENSE_KEY 
+            value: {{ .Values.pro.licenseKey }}
```

Can this be added as a secret instead?
```suggestion
            valueFrom:
              secretKeyRef:
                name: ambassador-pro-license-key
                key: key
```

Then just add this as a template
``` yaml
---
apiVersion: v1
kind: Secret
metadata:
    name: ambassador-pro-license-key
type: Opaque
data:
    key: {{ .Values.pro.licenseKey | b64enc }}
```

---

yes we can add the option to do both.